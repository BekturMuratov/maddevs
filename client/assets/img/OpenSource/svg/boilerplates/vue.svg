<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>