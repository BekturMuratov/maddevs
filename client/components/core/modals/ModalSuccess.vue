<template>
  <!-- id is needed for google analytics, don't remove it -->
  <Modal
    :id="id"
    ref="successModal"
  />
</template>

<script>
import Modal from '@/components/core/Modal'

export default {
  name: 'ModalSuccess',
  components: {
    Modal,
  },

  props: {
    id: {
      type: String,
      default: null,
    },
  },

  methods: {
    show() {
      this.$refs.successModal.makeSuccess()
      this.$refs.successModal.show()
    },

    close() {
      this.$refs.successModal.close()
    },
  },
}
</script>
