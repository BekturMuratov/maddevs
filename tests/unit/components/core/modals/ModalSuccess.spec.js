import { render } from '@testing-library/vue'
import { mount } from '@vue/test-utils'
import ModalSuccess from '@/components/core/modals/ModalSuccess'

const props = {
  visibled: true,
  id: 'modal-id',
  displayTime: 1000,
}

const $refs = {
  successModal: {
    makeSuccess: jest.fn(),
    show: jest.fn(),
    close: jest.fn(),
  },
}

describe('ModalSuccess component', () => {
  it('should render correctly', () => {
    const { container } = render(ModalSuccess, {
      props,
      $refs,
    })

    expect(container).toMatchSnapshot()
  })

  it('should modal correctly open and close', async () => {
    const wrapper = mount(ModalSuccess, {
      props,
    })

    wrapper.vm.show()

    expect(wrapper.findComponent({ name: 'SuccessMessage' })).toBeTruthy()

    wrapper.vm.close()

    expect(wrapper).toMatchSnapshot()
  })
})
