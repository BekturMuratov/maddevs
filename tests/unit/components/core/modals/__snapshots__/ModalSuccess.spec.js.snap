// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSuccess component should modal correctly open and close 1`] = `
<transition-stub name="main-fade" mode="out-in">
  <!---->
</transition-stub>
`;

exports[`ModalSuccess component should render correctly 1`] = `
<div>
  <transition-stub
    displaytime="1000"
    mode="out-in"
    name="main-fade"
    visibled="true"
  >
    <!---->
  </transition-stub>
</div>
`;
